From 15d4607a08964fbc98f95e691d32d2a1b1e3ce45 Mon Sep 17 00:00:00 2001
From: Matteo Bruni <mbruni@codeweavers.com>
Date: Mon, 27 Oct 2025 12:37:17 +0100
Subject: [PATCH] ntdll: Fix wait argument to NtWaitForMultipleObjects() from
 NtWaitForSingleObjects().

It's not going to matter in practice but it seems better to use the
correct argument, if nothing else to make sure that the wrong example
isn't accidentally copied around in the future.

Fixes: e6a6ae70682fcdc0b4e1573956fca163cbdff7a9
---
 dlls/ntdll/unix/sync.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/ntdll/unix/sync.c b/dlls/ntdll/unix/sync.c
index 07ffe854104..62c09c5e60b 100644
--- a/dlls/ntdll/unix/sync.c
+++ b/dlls/ntdll/unix/sync.c
@@ -2333,7 +2333,7 @@ NTSTATUS WINAPI NtWaitForMultipleObjects( DWORD count, const HANDLE *handles, WA
  */
 NTSTATUS WINAPI NtWaitForSingleObject( HANDLE handle, BOOLEAN alertable, const LARGE_INTEGER *timeout )
 {
-    return NtWaitForMultipleObjects( 1, &handle, FALSE, alertable, timeout );
+    return NtWaitForMultipleObjects( 1, &handle, WaitAll, alertable, timeout );
 }
 
 
-- 
GitLab

